src/main/java/org/jenkinsci/plugins/workflow/steps/EnvironmentExpander.java

```diff
@@ -45,6 +46,29 @@
     public abstract void expand(@Nonnull EnvVars env) throws IOException, InterruptedException;
 
     /**
+     * Provides an expander for a constant map of string keys and string values.
+     *
+     * @param env A non-null map of string keys and string values.
+     * @return An expander which will provide the given map.
+     */
+    public static EnvironmentExpander constant(@Nonnull Map<String,String> env) {
+        return new ConstantEnvironmentExpander(env);
+    }
+
+    private static class ConstantEnvironmentExpander extends EnvironmentExpander {
+        private static final long serialVersionUID = 1;
+        private final @Nonnull Map<String,String> envMap;
+
+        ConstantEnvironmentExpander(@Nonnull Map<String,String> envMap) {
+            this.envMap = envMap;
```

Comment:
Make a copy of the input map for safety (and in case the original does not serialize well).